submissions/Oleksandr-Heleta/OOP_exercise/index.js

```diff
@@ -14,13 +14,23 @@ class Inhabitants {
       this.gender = gender;
       this.legs = legs;
       this.say = say;
+      this.propertiesOrder = ['species', 'name', 'gender', 'legs', 'say']
+   }
+
+   addProperty(property, order) {
+      this.propertiesOrder.splice(order, 0, property);
+   }
```

Comment:
The following is still valid:

> The point of encapsulation principle of OOP is that every class handles its own properties.
> Let's implement class methods that produce class instance presentation string.

> You will employ method reloading and explicit inherited method version calling via super.

> The original project and task materials contain links to relevant sources that you will find helpful.

What is problem here? If a child class changes the way its property is implemented (e.g. we want to store now hands as an array of left and right hands) the code in a base class would break. This means we should change something in the base class to properly handle a property owned by a child class. This breaks the encapsulation principle. Let every class handle the properties it owns solely. JS provides required mechanisms. See hints in the comment above.